community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/RecordProcessorStep.java

```diff
@@ -59,4 +66,11 @@ protected void done()
         super.done();
         processor.done();
     }
+
+    @Override
+    protected void addStatsProviders( Collection<StatsProvider> providers )
+    {
+        super.addStatsProviders( providers );
+        providers.addAll( additionalStatsProviders );
```

Comment:
Here you add the StatsProviders to super, but in the constructor you don't send them up to the parent class? Oh wait, this method adds StatsProviders _from_ this instance _to_ the given collection. This method has a confusing name. I suggest adding "To" or "Into" to the end of the name to make it more clear.